### PR TITLE
Use MMI instead of MLJBase [WIP]

### PR DESCRIPTION
**Pull request by ayush-1506**
Should fix: #10

### CODE REVIEW

File: src/core.jl [hunk 1]
```diff
@@ -21,7 +21,7 @@ for opt in (:Descent, :Momentum, :Nesterov, :RMSProp, :ADAM, :AdaMax,
 
     @eval begin
 
-        MLJBase.istransparent(m::Flux.$opt) = true
+        #MLJModelInterface.istransparent(m::Flux.$opt) = true
```
Comment (ablaom) [inline src/core.jl]: extraneous `#`

**Comment by ablaom** [comment]
@ayush-1506 Thanks for that!

The fail on travis you had was a project compatibility issue, which I have fixed by swapping the MLJBase [compat] entry for a MLJModelInterface one. 

Yes, `istransparent` is not in MLJModelInterface but it should be.  I've opened an issue and added a TODO to the MLJFlux code. This is minor glitch, which means that `MLJ.params(some_mlj_flux_model)` does not expose the optimiser parameters, and none of this is needed here (or in tuning in any essential way).

The current fail on travis doesn't appear related to the purpose of this PR. Perhaps you added a test? I'll let you look into it at your leisure. Let me know if something else comes up.

**Comment by ablaom** [comment]
https://github.com/alan-turing-institute/MLJModelInterface.jl/issues/28

**Comment by ayush-1506** [comment]
@ablaom What's puzzling me is that I've not changed any test; In fact I've not touched any test file (I had previously made a small unwanted change, which I've now reverted), but the test still fails.

**Comment by ayush-1506** [comment]
The changes at the moment include replacing MLJBase by MMI everywhere in src/ , and small miscellaneous changes (like `schema`). My guess is that MMI is causing some unwanted trouble here.

**Comment by ablaom** [comment]
That's weird. I can only think that the switch to MMI has changed a relevant version of some dependency. A quick look shows that both travis tests use the same Flux and Tracker version. Maybe diff the Manifests to get some other clue. I also suggest you add a work-around for the "missing" method to see what happens:
```
ERROR: LoadError: LoadError: MethodError: no method matching Float32(::Tracker.TrackedReal{Float32})
```

**Pull request merged**